nixos/modules/services/security/cfssl.nix

```diff
@@ -0,0 +1,205 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.cfssl;
+in {
+  options.services.cfssl = {
+    enable = mkEnableOption ''
+      Whether to enable the CFSSL CA api-server.
+    '';
+
+    dataDir = mkOption {
+      default = "/var/lib/cfssl";
+      type = types.str;
+      description = "cfssl work directory";
+    };
+
+    address = mkOption {
+      default = "127.0.0.1";
+      type = types.str;
+      description = "Address to bind";
+    };
+
+    port = mkOption {
+      default = 8888;
+      type = types.int;
+      description = "Port to bind";
+    };
+
+    ca = mkOption {
+      default = "file:${cfg.dataDir}/ca.pem";
+      type = types.str;
+      description = "CA used to sign the new certificate -- accepts '[file:]fname' or 'env:varname'";
+    };
+
+    caKey = mkOption {
+      default = "${cfg.dataDir}/ca-key.pem";
+      type = types.str;
+      description = "CA private key -- accepts '[file:]fname' or 'env:varname'";
+    };
+
+    caBundle = mkOption {
+      default = null;
+      type = types.nullOr types.path;
+      description = "Path to root certificate store";
+    };
+
+    intBundle = mkOption {
+      default = null;
+      type = types.nullOr types.path;
+      description = "Path to intermediate certificate store";
+    };
+
+    intDir = mkOption {
+      default = null;
+      type = types.nullOr types.path;
+      description = "Intermediates directory";
+    };
+
+    metadata = mkOption {
+      default = null;
+      type = types.nullOr types.path;
+      description = ''
+        Metadata file for root certificate presence.
+        The content of the file is a json dictionary (k,v): each key k is
+        a SHA-1 digest of a root certificate while value v is a list of key
+        store filenames.
+      '';
+    };
+
+    remote = mkOption {
+      default = null;
+      type = types.nullOr types.str;
+      description = "Remote CFSSL server";
+    };
+
+    configFile = mkOption {
+      default = null;
+      type = types.nullOr types.str; # FIXME: types.nonStorePath, may contain secrets
```

Comment:
I think it's the user's responsibility to know when secrets would end up in the nix store. There are also a lot of users for which that isn't much of a problem anyways because they encrypt their nix store and are the only users.

A warning in the description would be appropriate though. Same for the other options that have a FIXME.